src/jit/optcse.cpp

```diff
@@ -2348,10 +2371,32 @@ class CSE_Heuristic
 
                 noway_assert(asg->gtOp.gtOp1->gtOper == GT_LCL_VAR);
 
+                // Backpatch the SSA def, if we're putting this cse temp into ssa.
+                asg->gtOp.gtOp1->AsLclVarCommon()->SetSsaNum(cseSsaNum);
```

Comment:
It would be better to use `AsLclVar` instead of `AsLclVarCommon` as you cannot have fields here (and there's actually an assert above checking just that).